table/sst_file_writer.cc

```diff
@@ -89,6 +96,12 @@ struct SstFileWriter::Rep {
     ikey.Set(user_key, sequence_number, value_type);
 
     builder->Add(ikey.Encode(), value);
+    Status s = validator->Add(user_key, value);
```

Comment:
This part needs to be `ikey.Encode()` instead of `user_key` (difference is https://github.com/facebook/rocksdb/blob/449029f865171507130290a9a4886e890f3cc39c/db/dbformat.h#L28-L29) since  `internal_comparator` is an `InternalKeyComparator`.